### DISCUSSION TITLE
Validate only the field that is firing the onBlur event

### DISCUSSION THREAD

**Discussion by st11x** [react-hook-form/react-hook-form#9874] [Q&A]
Hi,

I have a similar question to https://github.com/react-hook-form/react-hook-form/discussions/3699 but I am not using any validation library.

On my form are some input fields registered with custom validate methods that runs a fetch. For example, a username field registered with a custom validate function that runs a fetch. 

I set my useForm mode='onBlur'. 

When I am on a different field like the email field and tab out of it to the phone field, RHF invokes the validate method of the username field. as well.

Is there a setting to not have it do that? 

thanks
  Matt

**Comment by bluebill1049** [comment]
> When I am on a different field like the email field and tab out of it to the phone field, RHF invokes the validate method of the username field. as well.

not sure what do you mean, each field should validate itself unless used with `trigger`

**Reply by Moshyfawn** [reply]
Yea, has to be something you've done in your form, @st11x.

Here's [a working sandbox](https://codesandbox.io/s/jolly-mclaren-ideocr?file=%2Fsrc%2FApp.js): first field is "fetch-validated" and the second is client-side. The second field's validation doesn't get triggered

**Reply by st11x** [reply]
Thanks for the example, I found the cause but I couldn't explain it. In your example,

```
const { register, formState } = useForm({
```

if I destructure the formState object during useForm like this

```
const { register, formState : { errors, isValid} } = useForm({
```

then I'll get the problem. However, if I did it like you did on a separate line, then there's no issues.

In fact, **isValid** is the problem. the errors variable is fine.

**Reply by st11x** [reply]
It seems like isValid is the problem whether I declare it with the useForm or separately.
